sage/pallet/src/mock.rs

```diff
@@ -156,21 +171,179 @@ impl SageApi for SageMock {
+		Ok(SeasonConfig::<Self::Balance> {
+			fee: SeasonFeeConfig::<Self::Balance> {
+				transfer_asset: MockExistentialDeposit::get(),
+				buy_asset_min: MockExistentialDeposit::get(),
+				buy_percent: 1,
+				upgrade_asset_inventory: MockExistentialDeposit::get(),
+				unlock_trade_asset: MockExistentialDeposit::get(),
+				unlock_transfer_asset: MockExistentialDeposit::get(),
+				state_transition: MockExistentialDeposit::get(),
+			},
+		})
+	}
+}
+
+pub struct MockAffiliatesFeeProvider;
+
+impl FeeProvider for MockAffiliatesFeeProvider {
+	type AccountId = MockAccountId;
+	type FeeIdentifier = AffiliateMethods;
+	type FeeCurrency = MockBalance;
+	type FeeOutput = Vec<(MockBalance, MockAccountId)>;
+
+	fn get_fee_from(
+		_base_fee: Self::FeeCurrency,
+		_account: &Self::AccountId,
+		_identifier: &Self::FeeIdentifier,
+	) -> Self::FeeOutput {
+		Vec::with_capacity(0)
+	}
+}
+
+pub struct MockTournamentFeeProvider;
+
+impl FeeProvider for MockTournamentFeeProvider {
+	type AccountId = MockAccountId;
+	type FeeIdentifier = MockSeasonId;
+	type FeeCurrency = MockBalance;
+	type FeeOutput = (MockBalance, MockAccountId);
+
+	fn get_fee_from(
+		base_fee: Self::FeeCurrency,
+		account: &Self::AccountId,
+		_identifier: &Self::FeeIdentifier,
+	) -> Self::FeeOutput {
+		(base_fee, *account)
+	}
+}
+
+pub struct MockTreasuryManager;
+
+impl TreasuryManager for MockTreasuryManager {
+	type AccountId = MockAccountId;
+	type Currency = MockBalance;
+	type TreasuryPotKey = MockSeasonId;
+
+	fn is_treasurer_for(
+		_key: Self::TreasuryPotKey,
+		account: &Self::AccountId,
+	) -> Result<(), DispatchError> {
+		ensure!(account == &TREASURER, DispatchError::BadOrigin);
+		Ok(())
+	}
+
+	fn get_treasurer_for(_key: Self::TreasuryPotKey) -> Result<Self::AccountId, DispatchError> {
+		Ok(TREASURER)
+	}
+
+	#[cfg(feature = "runtime-benchmarks")]
+	fn set_treasurer_for(_key: Self::TreasuryPotKey, _owner: Self::AccountId) {
+		todo!()
+	}
+
+	fn deposit_into(
+		depository: &Self::AccountId,
+		_key: &Self::TreasuryPotKey,
+		fee: Self::Currency,
+	) -> Result<(), DispatchError> {
+		Balances::transfer(depository, &TREASURER, fee, ExistenceRequirement::KeepAlive)
+	}
+}
+
+pub struct MockTradeHandler;
+
+pub type MockTradeFilter = u32;
+
+impl TradeManager for MockTradeHandler {
+	type TradeFilter = MockTradeFilter;
+	type Asset = Asset;
+
+	fn is_tradeable_using(asset: &Self::Asset, filter: &Self::TradeFilter) -> bool {
+		asset.asset_type == *filter
+	}
+}
+
+pub type SageInstance1 = pallet_sage::Instance1;
+impl crate::Config<SageInstance1> for Test {
+	type PalletId = ExamplePalletId;
 	type SageGameTransition = ExampleTransitionGeneric<MockAccountId, MockBalance, SageMock>;
 	type SageApi = SageMock;
+	type SeasonHandler = MockSeasonManager;
+	type FeeHandler = GameFeeHandler<
+		MockAccountId,
+		Balances,
+		MockAffiliatesFeeProvider,
+		MockTournamentFeeProvider,
+		MockTreasuryManager,
+	>;
+	type TradeHandler = MockTradeHandler;
```

Comment:
What kind of customizations do we expect here? I am not sure if we need this type.

Reply:
We need it because the Asset's internal structure is not known to the SAGE pallet so we need an external source that actually knows about it to implement any filtering logic